src/Bicep.Core/SourceCode/SourceArchive.cs

```diff
@@ -91,17 +100,22 @@ public partial class SourceArchive // Partial required for serialization
 
         #endregion
 
-        #region Serialization
-
-        public partial record SourceFileInfo(
+        // This is the info we expose via SourceFiles
+        public record SourceFileInfo(
             // Note: Path is also used as the key for source file retrieval
             string Path,        // The location, relative to the main.bicep file's folder or one of the other roots.
-
             string ArchivePath, // The location (relative to root) of where the file is stored in the archive (munged from Path, e.g. in case Path starts with "../")
-            string Kind,        // Kind of source
-            string Contents     // File contents
+            string Kind,        // Kind of source (SourceKind)
+            string Contents,    // File contents
+            string? SourceArtifactId // Points to an external artifact that contains the source for this module (e.g. "br:contoso.io/test/module1:v1"), appears in v0.26 and higher
```

Comment:
How come you chose a nullable string vs an IArtifactReference object here to link back to the module?

Reply:
Good point.  61ffaca34f4809461276b4a8fa9d1331b70266cf